projects/wave-core/src/lib/operators/dialogs/data-repository/vector/vector-source-dataset.component.ts

```diff
@@ -96,6 +96,68 @@ export class VectorSourceDatasetComponent implements OnInit {
         });
         this.projectService.addLayer(l);
     }
+    // maybe refactor the following methods with central methods for finding min/max etc.
+    ngOnChanges(changes: SimpleChanges) {
+        for (const key in changes) {
+            if (changes.hasOwnProperty(key)) {
+                switch (key) {
+                    // check if there is any time-validity start/end data. If there is start/end data show the column of this data.
+                    // Also find a global Start/End (Min/Max) for the whole dataset.
+                    case 'dataset': {
+                        let globalDatasetMin = this.dataset.time_start;
+                        let globalDatasetMax = this.dataset.time_end;
+                        this.dataset.vectorLayer.forEach((element) => {
+                            for (const sourceVar in element) {
+                                if (element.hasOwnProperty(sourceVar)) {
+                                    if (sourceVar === 'time_start') {
+                                        if (element.time_start.isValid()) {
+                                            if (!this._displayedColumns.includes('start')) {
+                                                this._displayedColumns.push('start');
+                                            }
+                                            if (element.time_start < globalDatasetMin) {
+                                                globalDatasetMin = element.time_start;
+                                            }
+                                        }
+                                    }
+                                    if (sourceVar === 'time_end') {
+                                        if (element.time_end.isValid()) {
+                                            if (!this._displayedColumns.includes('end')) {
+                                                this._displayedColumns.push('end');
+                                            }
+                                            if (element.time_end > globalDatasetMax) {
+                                                globalDatasetMax = element.time_end;
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        });
+                        // Fill in the global Min/Max for rows without Start/End
+                        this.dataset.vectorLayer.forEach((element) => {
+                            for (const sourceVar in element) {
+                                if (element.hasOwnProperty(sourceVar)) {
+                                    if (sourceVar === 'time_start') {
+                                        if (!element.time_start.isValid()) {
+                                            element.time_start = globalDatasetMin;
```

Comment:
This is unfortunately incorrect. The rule is: If the layer has no own time interval, use the one from the dataset.
If you changed the dataset's validity before to display the global min/max, then this potentially produces wrong intervals.

Reply:
i hope its fixed in 6da1d4c74b7a75a702074ebf5ef83c4bda092d4f

Reply:
reverted the dataset manipulation/processing part in bcf99b0df6ed19d67ce802b25df192cf7edd124e

As discussed i would like to do the dataset processing part in a new PR